### ISSUE TITLE
how to convert to base 64 and DATA URI.

### ISSUE THREAD

**Issue by Divyaganapathy** [wymsee/cordova-imagePicker#116]
controller 
 $scope.choosePhotos = function () {
    var options = {
      maximumImagesCount: 10,
      quality: 80,
      targetWidth: 1000,
      targetHeight: 1000,
        };

    $cordovaImagePicker.getPictures(options).then(function (results) {
    for (var i = 0; i < results.length; i++) {
    $scope.PhotoData.push(results[i]);
    console.log(results[i]);
            }
        if(!$scope.$$phase) {
          $scope.$apply();
          }
            }, function (err) {
                        // An error occured. Show a message to the user
                    });
                    };

html 
<div class="list">
      <div class="item">
        <ion-slide-box>
					<ion-slide ng-repeat="item in PhotoData">
        <img ng-src="data:image/jpg;base64,{{item}}" style="max-width: 100%">
      </ion-slide>
      				</ion-slide-box>
        </div>
      </div>

**Comment by Divyaganapathy** [comment]
<div class="list">
      <div class="item">
        <ion-slide-box>
					<ion-slide ng-repeat="item in PhotoData">
        <img ng-src="data:image/jpg;base64,{{item}}" style="max-width: 100%">
      </ion-slide>
      				</ion-slide-box>
        </div>
      </div>

**Comment by chrisbenseler** [comment]
I'm having the same problem; the plugin outputs the file in a "content://com.android.providers.media.documents/document/{anything}" path and Android doesn't allow to access files from content:// in order to make it a base64 file

**Comment by urbanmania** [comment]
If you guys have solved this I'd like to know how. My solution to draw them on canvas and converting them that way is resulting in huge images

**Comment by firescout** [comment]
just managed to do this.
might now be the most elegant way

`imageBase64 = function (url, callback, outputFormat) {
        var canvas = document.createElement('CANVAS'),
                ctx = canvas.getContext('2d'),
                img = new Image;
        img.crossOrigin = 'Anonymous';
        img.onload = function () {
            var dataURL;
            canvas.height = img.height;
            canvas.width = img.width;
            ctx.drawImage(img, 0, 0);
            dataURL = canvas.toDataURL(outputFormat); // the base64 data of the image
            dataURL = dataURL.substring("data:image/png;base64,".length);
            callback.call(this, dataURL);
            canvas = null;
        };
        img.src = url;
    }`

then 
`selectedImage.imageBase64(results[0], function (base64Img) {
               console.log(base64Img);
            });`

**Comment by urbanmania** [comment]
yes, this works, but the problem with is it you take for example a compressed jpeg and convert it to an uncompressed png, resulting in a double the size of the file or more.

**Comment by tomercagan** [comment]
You can use FileReader:

```
function processImage(file) {
    fileReader = new FileReader();
    fileReader.onload = onload;
    fileReader.onprogress = progress;
    resolveFile(file).then(function(actualFile) {
        fileReader.readAsBinaryString(actualFile);
    });
}

var resolveFile = function(entry) {
    var deferred = $q.defer();
    // first convert to local file system URL
    window.resolveLocalFileSystemURL(entry, function(fileEntry) {
        // now read/convert the file to file object.
        fileEntry.file(function(file) {
            console.log("File converted to file entry");
            deferred.resolve(file);
        }, function(err) {
            console.log("Failed to convert to file entry", err);
            deferred.reject(err);
        });
    }, function(err) {
        console.log("Failed to resolve to file URL", err);
        deferred.reject(err);
    });

    return deferred.promise;
};

var onload = function (evt) {
    var encoded = window.btoa(evt.target.result);
    // now encoded contains the base64 image.
    // note it is without the "header"
    // img.src = "data:image/jpeg;base64," + encoded
};
```

Initially I tried to use `fileReader.readAsDataURL(actualFile);` which give you the base64 data in one shot but it resulted in corrupt images. A workaround was suggested with `readAsBinaryString()` and then `window.btoa()` and solved it. See [SO question here](http://stackoverflow.com/questions/37392303/cordova-reading-large-image-corrupts-image).
